docs/src/developers_guide/contributing_tests.rst

```diff
@@ -0,0 +1,301 @@
+------------------------------------------------------------------------------------
+
+As far as is possible, the actual test function should do little else but the
+actual assertion. Separating off preparation into fixtures may make the code
+harder to follow, so compromises are acceptable. For example, setting up a test
+``Cube`` should be a fixture, whereas creating a simple string
+(``expected = "foo"``), or a single use setup, should *not* be a fixture.
+
+
+New fixtures should always be considered for conftest when added. If it is
+decided that they are not suitably reusable, they can be placed within the
+local test file.
+
+`Parameterisation <https://docs.pytest.org/en/stable/example/parametrize.html>`_
+--------------------------------------------------------------------------------
+
+Though it is a useful tool, we should not be complicating tests to work around
+parameters; they should only be used when it is simple and apparent to implement.
+
+Where you are parameterising multiple tests with the same parameters, it is
+usually prudent to use the `parameterisation within fixtures
+<https://docs.pytest.org/en/stable/how-to/fixtures.html#parametrizing-fixtures>`_.
+When doing this, ensure within the tests that it is apparent that they are being
+parameterised, either within the fixture name or with comments.
+
+All parameterisation benefits from
+`ids <https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids>`_,
+and so should be used where possible.
+
+`Classes <https://docs.pytest.org/en/stable/getting-started.html#group-multiple-tests-in-a-class>`_
+---------------------------------------------------------------------------------------------------
+
+How and when to group tests within classes can be based on personal opinion,
+we do not deem consistency on this a vital concern.
+
+`Mocks <https://docs.pytest.org/en/stable/how-to/monkeypatch.html>`_
+--------------------------------------------------------------------
+
+Any mocking should be done with ``pytest.mock``, and monkeypatching where suitable.
+
+.. note::
+    If you think we're missing anything important here, please consider creating an
+    issue or discussion and share your ideas with the team!
+
+.. _pytesting_tools:
+
+=============
+Testing tools
+=============
+
+.. note::
+    :class:`iris.tests.IrisTest` has been deprecated, and replaced with
+    the :mod:`iris.tests._shared_utils` module.
+
+Iris has various internal convenience functions and utilities available to
+support writing tests. Using these makes tests quicker and easier to write, and
+also consistent with the rest of Iris (which makes it easier to work with the
+code). Most of these conveniences are accessed through the
+:mod:`iris.tests._shared_utils` module.
+
+.. tip::
+
+    All functions listed on this page are defined within
+    :mod:`iris.tests._shared_utils`. They can be accessed within a test using
```

Comment:
I think we decided that we couldn't commit to publicly maintained API at this time, hence having `_shared_utils` private. However it does make this documentation harder to read, since references to API objects (e.g. `assert_logs()`) cannot link to anything in the docs. I would definitely be up for making a public module, analogous to `np.testing` (as opposed to `np.tests`), with documented conveniences that anyone can use.

If you agree, I will raise an issue.

Reply:
#6210